packages/filesystem/src/common/filesystem.ts

```diff
@@ -95,6 +95,11 @@ export interface FileSystem extends Disposable {
      */
     getRoots(): Promise<FileStat[]>;
 
+    /**
+     * Returns a promise the resolves to a file stat representing the current user's home directory.
+     */
+    getCurrentUserHome(): Promise<FileStat>;
```

Comment:
I like this api, I think both preferences and keymaps could use that.

Reply:
I really would like to be able to store user settings on a central server in a database. Which is why I asked for the user-settings service.